systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectIsolatedST.java

```diff
@@ -1318,6 +1319,31 @@ void testKafkaConnectWithScramShaAuthenticationRolledAfterPasswordChanged(Extens
         KafkaConnectUtils.waitUntilKafkaConnectRestApiIsAvailable(namespaceName, kafkaConnectPodNameAfterRU);
     }
 
+    @ParallelNamespaceTest
+    void testConnectRackAwareness(ExtensionContext extensionContext) {
+        TestStorage storage = new TestStorage(extensionContext);
+        String topologyKey = "kubernetes.io/hostname";
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(storage.getClusterName(), 1, 1).build());
+        resourceManager.createResource(extensionContext, KafkaConnectTemplates.kafkaConnect(extensionContext, storage.getClusterName(), 1)
+                .editSpec()
+                    .withNewRack(topologyKey)
+                .endSpec()
+                .build());
+
+        LOGGER.info("Connect cluster deployed successfully");
+        String podName = PodUtils.getPodNameByPrefix(storage.getNamespaceName(), KafkaConnectResources.deploymentName(storage.getClusterName()));
+        String podJson = TestUtils.toJsonString(kubeClient(storage.getNamespaceName()).getPod(podName));
+        assertThat(podJson, hasJsonPath(StUtils.initGlobalVariableJsonPathBuilder(0, "RACK_TOPOLOGY_KEY"), hasItem(topologyKey)));
+        assertThat(podJson, hasJsonPath(StUtils.nodeSelectorKeyJsonPathBuilder(topologyKey), hasItem(topologyKey)));
+
+        String podLogs = kubeClient(storage.getNamespaceName()).logs(podName);
+        String podNodeName = PodUtils.getPodByName(storage.getNamespaceName(), podName).getSpec().getNodeName();
+        String hostname = podNodeName.contains(".") ? podNodeName.substring(0, podNodeName.indexOf(".")) : podNodeName;
+        String clientRackConfig = "consumer.client.rack=" + hostname;
+        assertTrue(podLogs.contains(clientRackConfig), "Pod logs contain " + clientRackConfig);
+    }
+
```

Comment:
This is already tested in `testRackAwareConnectCorrectDeployment`, so you can remove it.

Reply:
That method is not checking `consumer.client.rack` configuration, which is the core of this feature. We can integrate that part in `testRackAwareConnectCorrectDeployment`. WDYT?

TBH the test class name doesn't make it easy to find out that it is dealing with rack awareness. I would suggest `RackAwarenessIsolatedST`.

Reply:
Yeah I agree that name is not making it easy to find it out. The `SpecificIsolatedST` has more then just rack awareness tests, so maybe moving the tests to separate class and renaming the test would make it clear?

Reply:
Yeah I'll try to do that.

Reply:
If you'll need any help, feel free to ping me :)